### PR TITLE
Add wien2k parser

### PR DESCRIPTION
**Pull request by saurabh02**

### CODE REVIEW

File: dftparse/wien2k/eloss_parser.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            "eloss$_{zz}$": eloss_zz}
+
+
+base_rules = [
+    (lambda x: "# Energy [eV]    eloss_xx      eloss_zz" in x, _parse_eloss)
+]
+
+
+class ElossParser(BlockParser):
+    """Parser for Wien2k's .absorp file"""
```
Comment (maxhutch) [inline dftparse/wien2k/eloss_parser.py]: update

File: dftparse/wien2k/epsilon_parser.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            "Im $\\varepsilon_{zz}$": im_eps_zz}
+
+
+base_rules = [
+    (lambda x: "# Energy [eV] Re_eps_xx     Im_eps_xx     Re_eps_zz     Im_eps_zz" in x, _parse_epsilon)
+]
+
+
+class EpsilonParser(BlockParser):
+    """Parser for Wien2k's .absorp file"""
```
Comment (maxhutch) [inline dftparse/wien2k/epsilon_parser.py]: update

File: dftparse/wien2k/absorp_parser.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    re_sigma_xx = []
+    re_sigma_zz = []
+    absorp_xx = []
+    absorp_zz = []
+
+    while newline[0] != "#" and len(newline.split()) == 5:
+        split_line = newline.split()
+
+        energies.append(float(split_line[0]))
+        wavelengths.append(float(split_line[0])/1240)
```
Comment (maxhutch) [inline dftparse/wien2k/absorp_parser.py]: Why divide by 1240?

File: dftparse/wien2k/absorp_parser.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        frequencies.append(float(split_line[0])*2.418*10**14)
```
Comment (maxhutch) [inline dftparse/wien2k/absorp_parser.py]: Why multiply by `2.418*10**14`?

File: dftparse/wien2k/reflectivity_parser.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            "reflect$_{zz}$": reflect_zz}
+
+
+base_rules = [
+    (lambda x: "# Energy [eV]  reflect_xx    reflect_zz" in x, _parse_reflectivity)
+]
+
+
+class ReflectivityParser(BlockParser):
+    """Parser for Wien2k's .absorp file"""
```
Comment (maxhutch) [inline dftparse/wien2k/reflectivity_parser.py]: update

File: dftparse/wien2k/refract_parser.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            "ref_ind$_{zz}$": ref_ind_zz, "extinct$_{xx}$": extinct_xx, "extinct$_{zz}$": extinct_zz}
+
+
+base_rules = [
+    (lambda x: "# Energy [eV]  ref_ind_xx    ref_ind_zz    extinct_xx    extinct_zz" in x, _parse_refraction)
+]
+
+
+class RefractionParser(BlockParser):
+    """Parser for Wien2k's .refract file"""
```
Comment (maxhutch) [inline dftparse/wien2k/refract_parser.py]: Update

File: dftparse/wien2k/scf2_parser.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+
+base_rules = [
+    (lambda x: ":GAP (global)" in x, _parse_bandgap)
+]
+
+
+class Scf2Parser(BlockParser):
+    """Parser for Wien2k's .scf file"""
```
Comment (maxhutch) [inline dftparse/wien2k/scf2_parser.py]: update

File: dftparse/wien2k/absorp_parser.py (hunk 3 above)
Comment (saurabh02) [reply dftparse/wien2k/absorp_parser.py]: to convert energy to wavelength

File: dftparse/wien2k/absorp_parser.py (hunk 4 above)
Comment (saurabh02) [reply dftparse/wien2k/absorp_parser.py]: to convert energy to frequency

File: dftparse/wien2k/scf2_parser.py (hunk 7 above)
Comment (saurabh02) [reply dftparse/wien2k/scf2_parser.py]: update what here?

**Comment by saurabh02** [comment]
Would something like the last commit work?

File: dftparse/wien2k/scf2_parser.py (hunk 7 above)
Comment (maxhutch) [reply dftparse/wien2k/scf2_parser.py]: This is the parser for the scf2 file, right?

File: dftparse/wien2k/refract_parser.py (hunk 6 above)
Comment (maxhutch) [reply dftparse/wien2k/refract_parser.py]: Sorry, this one is already right.

File: dftparse/wien2k/reflectivity_parser.py (hunk 5 above)
Comment (maxhutch) [reply dftparse/wien2k/reflectivity_parser.py]: Is the reflectivity in the `absorp` file?

File: dftparse/wien2k/absorp_parser.py (hunk 3 above)
Comment (maxhutch) [reply dftparse/wien2k/absorp_parser.py]: That type of conversion shouldn't be in `dftparse`; you could do it in `pif-dft`.  `dftparse` is just for extracting structured data from the text files.

File: dftparse/wien2k/refract_parser.py (hunk 6 above)
Comment (saurabh02) [reply dftparse/wien2k/refract_parser.py]: ahh this is what threw me off

File: dftparse/wien2k/absorp_parser.py (hunk 3 above)
Comment (saurabh02) [reply dftparse/wien2k/absorp_parser.py]: ahh okay, i'll remove these and the attached tests

**Pull request merged**